### ISSUE TITLE
twisted.python.log.err does not have useful docstring

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2567]
|<img alt="konrads's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| konrads reported|
|-|-|
|Trac ID|trac#2567|
|Type|defect|
|Created|2007-04-12 13:35:08Z|
|Branch|https://github.com/twisted/twisted/tree/err-docstring-2567|

twisted.python.log.err does not have useful docstring.
It does not describe what arguments __stuff_ and more importantly __why_ are used for.

This is confusing.

<details><summary>Searchable metadata</summary>

```
trac-id__2567 2567
type__defect defect
reporter__konrads konrads
priority__normal normal
milestone__ 
branch__branches_err_docstring_2567 branches/err-docstring-2567
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__log__documentation log, documentation
time__1176384908000000 1176384908000000
changetime__1220270673000000 1220270673000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [24536]) Branching to 'err-docstring-2567'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

The text describing `_stuff` should just be part of the epytext markup for that parameter.  Some more text for the `_why` parameter would probably be helpful as well.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

Replying to [exarkun](#note_5):
> The text describing `_stuff` should just be part of the epytext markup for that parameter.  Some more text for the `_why` parameter would probably be helpful as well.

I moved the text in r24636, but leave the description for __why_ up to you since that is what the creator of this ticket asked for (but didnt provide), I'm just as confused as him ;)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun removed owner|
|-|-|

I added r24650

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_8"></a>|@glyph set owner to @exarkun|
|-|-|

Looks good; but could `_stuff` get a `@type` as long as we're at it?

Do that and merge it.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra commented|
|-|-|

there's one ```None``` left that is missing the C{}

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun commented|
|-|-|

(In [24658]) add @type for _stuff

refs [#2567](https://github.com/twisted/twisted/issues/2567)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set status to `closed`|
|-|-|

(In [24659]) Merge err-docstring-2567

Author: thijs, exarkun
Reviewer: glyph
Fixes: [#2567](https://github.com/twisted/twisted/issues/2567)

Improve the docstring documentation of the parameters to
`twisted.python.log.err`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
